python/scripts/filter_osm.py

```diff
@@ -12,111 +12,132 @@ class MissingInputError(Exception):
         command = ["osmium", "extract", "-c", coords, input_file]
     else:
-        command = None
+        raise InvalidInputError("Invalid coordinates or config file.")
 
     if strategy:
         command.extend(["-s", strategy])
     
-    if command:
-        subprocess.run(command)
-    else:
-        raise InvalidInputError("Invalid coordinates or config file.")
+    subprocess.run(command)
 
-if __name__ == '__main__':
-    # If no tag is used OR script is called with -h/--help
-    if len(sys.argv) < 2 or (tag:=sys.argv[1]) in ("-h", "--help"):
-        display_help()
-    elif tag == "-id":
-        if len(sys.argv) < 4:
-            raise MissingInputError("You need to specify relation ID and input file.")
-
-        relation_id = sys.argv[2]
-        input_file = sys.argv[3]
-        if not is_valid_extension(input_file):
-            raise InvalidInputError(f"File must have one of the following extensions: osm, osm.pbf, osm.bz2")
-        if len(sys.argv) == 5 and sys.argv[4] == '-s':
-            raise MissingInputError("Missing specified strategy type.")
-        strategy = sys.argv[5] if len(sys.argv) > 5 and sys.argv[4] == "-s" else None
-        
-        if strategy and not check_strategy(strategy):
-            raise InvalidInputError("Invalid strategy type. Call script.py -h/--help to display help.")
+def run_osmium_filter(input_file: str, expression_file: str, omit_referenced: bool):
+    """Function to filter objects based on tags in expression file.
+    Nodes referenced in ways and members referenced in relations will not 
+    be added to output if omit_referenced set to True.
+    """
+    cmd = ["osmium", "tags-filter", input_file, "-e", expression_file, "-o", "filtered.osm.pbf"]
+    if omit_referenced:
+        cmd.extend(["-R"])
+    subprocess.run(cmd)
+
+def parse_args(arg_list: list[str] | None):
+    parser = argparse.ArgumentParser(description="Filter OSM files with various operations.", formatter_class=argparse.RawTextHelpFormatter)
+
+    parser.add_argument("tag", choices=["id", "b", "f"])
+    parser.add_argument('input_file', nargs='?', help='Path to input OSM file')
+    parser.add_argument("-e", dest="expression_file", nargs="?", help="Path to expression file for filtering tags (required for 'f' tag)")
+    parser.add_argument("-c", dest="coords", nargs="?",help="Bounding box coordinates or path to config file (required for 'b' tag)")
+    parser.add_argument("-r", dest="relation_id", nargs="?", help="relation ID (required for 'b' tag)")
+    parser.add_argument("-s", dest="strategy", help="Strategy type (optional for 'id', 'b' tags)")
+    parser.add_argument("-R", dest="omit_referenced", action="store_true", help="Omit referenced objects (optional for 'f' tag)")
+
+    parser.format_help = lambda: display_help()
+    args = parser.parse_args(arg_list)
+
+    return args
+
+def main(arg_list: list[str] | None = None):
+    args = parse_args(arg_list)
+
+    if not args.input_file:
```

Comment:
This check (input_file is set) should be possible to do with the library. Usually, the parsing libraries allow the marking of specific args as mandatory and throw an exception if not set (probably it already does that already).